txnsync/exchange.go

```diff
@@ -56,6 +56,7 @@ type packedTransactionGroups struct {
 	_struct struct{} `codec:",omitempty,omitemptyarray"`
 
 	Bytes []byte `codec:"g,allocbound=maxEncodedTransactionGroupBytes"`
+	Compressed bool `codec:"c"`
```

Comment:
Maybe replace the "Compressed" with "CompressionFormat" of type uint8 ?
This would give us higher control over the format in the future.
( leave the value 0 for non-compress, and value of 1 for gzip )